### DISCUSSION TITLE
How to return response that downloads file using the Storage facade?

### DISCUSSION THREAD

**Discussion by drewgallagher** [rebing/graphql-laravel#1160] [Q&A] [answered]
Is there a way to return from GraphQL in a mutation or query a response that downloads a file using the storage facade? 
https://laravel.com/docs/11.x/filesystem#downloading-files

**Comment by mfn** [comment]
You can try and return anything you want, but if you use a proper GraphQL client, it probably won't be happy about it (it expects a certain JSON response format).

I think the proper most way is to return an URL to your _other_ endpoint to stream that file.

**Reply by drewgallagher** [reply]
Can you elaborate on this? I'm not sure what this means. 

"I think the proper most way is to return an URL to your other endpoint to stream that file."

Does this mean do not use the graphql endpoint for this and just use a Laravel route to return the response?

**Reply by mfn** [reply, marked as answer]
Yes
